crates/runtime/src/component/mod.rs

```diff
@@ -140,6 +140,23 @@ fn find_first_delimiter(from: &str) -> Option<(usize, usize)> {
     }
 }
 
+/// Enum representing the initialization type of a component.
+///
+/// [`OnStartup`] indicates that the component should be initialized when runtime started.
+/// [`OnTrigger`] indicates that the component should be initialized when a specific trigger event occurs.
+#[derive(Debug, Clone)]
```

Comment:
If you derive `Copy` then you won't need to clone so much in the rest of the code.